### PR TITLE
Use SDK waiters in UI tests

### PR DESCRIPTION
**Pull request by abrooksv**
Also, close clients, make sure we always delete the S3 bucket

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/S3BrowserTest.kt [hunk 1]
```diff
@@ -173,69 +180,41 @@ class S3BrowserTest {
...
+            s3Client.deleteBucket { it.bucket(bucket) }
+            waitForS3BucketDeletion()
+        } catch (e: Exception) {
+            if (e is NoSuchBucketException) {
+                return
             }
+            log.error("Delete bucket stack threw an exception", e)
         }
+
+        s3Client.close()
```
Comment (hunterwerlla) [inline ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/S3BrowserTest.kt:209]: put closes in finally?

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/CloudFormationBrowserTest.kt [hunk 2]
```diff
@@ -178,6 +172,8 @@ class CloudFormationBrowserTest {
             log.error("Delete stack threw an exception", e)
         }
         waitForStackDeletion()
+
+        cloudFormationClient.close()
```
Comment (hunterwerlla) [inline ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/CloudFormationBrowserTest.kt:176]: ditto

**Review by hunterwerlla — APPROVED** [review]

File: ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/S3BrowserTest.kt (hunk 1 above)
Comment (abrooksv) [reply ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/S3BrowserTest.kt]: Good call out, added

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**